assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -147,4 +150,48 @@ describe("DetourMap", () => {
     expect(screen.getByTitle("Detour Start")).not.toBeNull()
     expect(screen.queryByTitle("Detour End")).toBeNull()
   })
+
+  test("missed stops are filled in when detour is complete", async () => {
+    const stop = stopFactory.build()
+    jest.mocked(fetchDetourMissedStops).mockReturnValue(instantPromise([stop]))
+
+    const { container } = render(<DiversionPage />)
+
+    await fireEvent.click(
+      container.querySelector(".c-detour_map--original-route-shape")!
+    )
+
+    await fireEvent.click(
+      container.querySelector(".c-detour_map--original-route-shape")!
+    )
+
+    await waitFor(() =>
+      expect(screen.queryByText(stop.name)).toBeInTheDocument()
+    )
+
+    expect(screen.getByText(stop.name)).toBeInTheDocument()
+  })
+
+  test("duplicate missed stops are only rendered once", async () => {
+    const stop = stopFactory.build()
+    jest
+      .mocked(fetchDetourMissedStops)
+      .mockReturnValue(instantPromise([stop, stop]))
+
+    const { container } = render(<DiversionPage />)
+
+    await fireEvent.click(
+      container.querySelector(".c-detour_map--original-route-shape")!
+    )
+
+    await fireEvent.click(
+      container.querySelector(".c-detour_map--original-route-shape")!
+    )
+
+    await waitFor(() =>
+      expect(screen.queryByText(stop.name)).toBeInTheDocument()
+    )
+
+    expect(screen.getAllByText(stop.name)).toHaveLength(1)
+  })
```

Comment:
https://github.com/mbta/skate/pull/2419#discussion_r1503316396

```suggestion
    await waitFor(() =>
      expect(screen.getAllByText(stop.name)).toHaveLength(1)
    )
  })
```